ibc-clients/cw-context/src/context/client_ctx.rs

```diff
@@ -195,3 +196,37 @@ where
         Ok(())
     }
 }
+
+pub trait CwClientValidation<'a>: ClientValidationContext {
+    fn cosmwasm_query_context(&self) -> Option<&Deps<'a>>;
+    fn cosmwasm_execute_context(&mut self) -> Option<&mut DepsMut<'a>>;
+    fn generate_sha256_digest(&self, data: &[u8]) -> Checksum;
+}
+
+pub trait CwClientExecution<'a>: CwClientValidation<'a> + ClientExecutionContext {}
+
+impl<'a, C: ClientType<'a>> CwClientValidation<'a> for Context<'a, C>
+where
+    <C::ClientState as TryFrom<Any>>::Error: Display,
+    <C::ConsensusState as TryFrom<Any>>::Error: Display,
+{
+    fn cosmwasm_query_context(&self) -> Option<&Deps<'a>> {
+        self.deps.as_ref()
+    }
+
+    fn cosmwasm_execute_context(&mut self) -> Option<&mut DepsMut<'a>> {
+        self.deps_mut.as_mut()
+    }
+
+    fn generate_sha256_digest(&self, data: &[u8]) -> Checksum {
```

Comment:
@ljoss17, by the way, is it necessary to be having this method here? Couldn't we just use `Checksum::generate(data)` in the SN light client where this method is called?

Reply:
I see now. My suggestion is:
Since this is being used temporarily for the 2nd demo there, it’s probably better to place this line of code in the `ibc-client-starknet` crate for now and import it into the `ibc-client-starknet-cw`. So, we can keep things clean here.
Regarding the size limit, I think we’re going to hit that limit eventually as we work towards completing the Starknet light client implementation.

Reply:
Ok, I will close the PR in the meantime, and we can maybe keep the patch in ibc-starknet, https://github.com/informalsystems/ibc-starknet/blob/main/light-client/Cargo.toml#L64 ?

Reply:
Alright, let's go with a feature branch as Rano suggested and open ibc-starknet-related PRs against it. Once things consolidated, we can decide on whether to merge this feature branch into the main.